nntrainer/layers/nnstreamer_layer.cpp

```diff
@@ -81,13 +81,16 @@ static int nnst_info_to_tensor_dim(ml_tensors_info_h &out_res, TensorDim &dim) {
   if (type != ML_TENSOR_TYPE_FLOAT32)
     return ML_ERROR_NOT_SUPPORTED;
 
-  if (ML_TENSOR_RANK_LIMIT > ml::train::TensorDim::MAXDIM)
-    return ML_ERROR_NOT_SUPPORTED;
-
   status = ml_tensors_info_get_tensor_dimension(out_res, 0, dim_);
   if (status != ML_ERROR_NONE)
     return status;
 
+  if (ML_TENSOR_RANK_LIMIT > ml::train::TensorDim::MAXDIM) {
+    for (uint i = ml::train::TensorDim::MAXDIM; i < ML_TENSOR_RANK_LIMIT; i++)
+      if (dim_[i] > 1)
```

Comment:
If it accepts iff dim_[i]==1,
how about fixing it as `dim_[I] != 1` ?

Reply:
NNStreamer says the default value of the tensor dimension is 1. On second thought, it might be difficult to idendentify rank 5 with 1 last like ``` 2:3:4:5:1```. It is not the right solution then.